pkg/policyconditions/conditions.go

```diff
@@ -155,29 +157,55 @@ func update(apiWriter client.Client, apiReader client.Reader, policyReader clien
 			enactmentsCountByCondition.Aborted()
 
 		logger.Info(
-			fmt.Sprintf("numberOfNmstateMatchingNodes: %d, enactments count: %s", numberOfNmstateMatchingNodes, enactmentsCountByCondition),
+			fmt.Sprintf("numberOfNmstateMatchingNodes: %d, enactments count: %s",
+				numberOfNmstateMatchingNodes,
+				enactmentsCountByCondition),
 		)
 
 		if numberOfNmstateMatchingNodes == 0 {
 			message := "Policy does not match any node"
 			SetPolicyNotMatching(&policy.Status.Conditions, message)
 		} else if enactmentsCountByCondition.Failed() > 0 || enactmentsCountByCondition.Aborted() > 0 {
-			message := fmt.Sprintf("%d/%d nodes failed to configure", enactmentsCountByCondition.Failed(), numberOfNmstateMatchingNodes)
+			message := fmt.Sprintf(
+				"%d/%d nodes failed to configure",
+				enactmentsCountByCondition.Failed(),
+				numberOfNmstateMatchingNodes,
+			)
 			if enactmentsCountByCondition.Aborted() > 0 {
-				message += fmt.Sprintf(", %d nodes aborted configuration", enactmentsCountByCondition.Aborted())
+				message += fmt.Sprintf(
+					", %d nodes aborted configuration",
+					enactmentsCountByCondition.Aborted(),
+				)
 			}
 			SetPolicyFailedToConfigure(&policy.Status.Conditions, message)
-		} else if numberOfFinishedEnactments < numberOfNmstateMatchingNodes {
+		} else if numberOfFinishedEnactments < numberOfReadyNmstateMatchingNodes {
+			message := fmt.Sprintf(
+				"Policy is progressing %d/%d nodes finished",
+				numberOfFinishedEnactments,
+				numberOfReadyNmstateMatchingNodes,
+			)
+			if numberOfNotReadyNmstateMatchingNodes != 0 {
+				message += fmt.Sprintf(
+					", %d nodes ignored due to NotReady state",
+					numberOfNotReadyNmstateMatchingNodes,
+				)
+			}
```

Comment:
Can we create a little closure with this before the if statement to reuse things ?

Reply:
Thanks, done